src/plugins/intel_gpu/src/graph/include/fully_connected_inst.h

```diff
@@ -53,7 +53,11 @@ class typed_primitive_inst<fully_connected> : public typed_primitive_inst_base<f
     typed_primitive_inst(network& network, fully_connected_node const& node);
 
     memory::ptr weights_memory() const {
-        return is_dynamic() && _impl_params->reordered_weights != nullptr ? _impl_params->reordered_weights : dep_memory_ptr(1);
+        if (is_dynamic() && _reordered_weights_cache.size() != 0) {
+            return _reordered_weights_cache.get_recent_element().second;
```

Comment:
Reason why recent is used here is not that  obvious to my mind. What do you think about something like that:
```cpp
// in update_weights
impl_params.weights_layout = requested_weights_layout


if (is_dynamic()) {
   auto dyn_weights = _reordered_weights_cache.get(_impl_params.weights_layout);
   return dyn_weights ? dyn_weights : dep_memory(1)
} else { ... }

```